examples/vision/cutmix.py

```diff
@@ -0,0 +1,383 @@
+
+y_train = pd.get_dummies(y_train.flatten()).values.astype(np.float32)
+y_test = pd.get_dummies(y_test.flatten()).values.astype(np.float32)
+
+class_names = [
+    "Airplane",
+    "Automobile",
+    "Bird",
+    "Cat",
+    "Deer",
+    "Dog",
+    "Frog",
+    "Horse",
+    "Ship",
+    "Truck",
+]
+
+"""
+## Define hyperparameters
+"""
+
+AUTO = tf.data.experimental.AUTOTUNE
+BATCH_SIZE = 32
+IMG_SHAPE = 32
+
+"""
+## Dataloader
+"""
+
+
+def preprocess_image(image, label):
+    image = tf.image.resize(image, (IMG_SHAPE, IMG_SHAPE))
+    image = tf.image.convert_image_dtype(image, tf.float32) / 255.0
+    return image, label
+
+
+"""
+## Convert the data into TensorFlow `Dataset` objects
+"""
+
+train_ds_one = (
+    tf.data.Dataset.from_tensor_slices((x_train, y_train))
+    .shuffle(1024)
+    .map(preprocess_image, num_parallel_calls=AUTO)
+)
+train_ds_two = (
+    tf.data.Dataset.from_tensor_slices((x_train, y_train))
+    .shuffle(1024)
+    .map(preprocess_image, num_parallel_calls=AUTO)
+)
+
+train_ds_simple = tf.data.Dataset.from_tensor_slices((x_train, y_train))
+
+test_ds = tf.data.Dataset.from_tensor_slices((x_test, y_test))
+
+train_ds_simple = (
+    train_ds_simple.map(preprocess_image, num_parallel_calls=AUTO)
+    .batch(BATCH_SIZE)
+    .prefetch(AUTO)
+)
+
+# We will be combining two shuffled datasets from the same training data.
+train_ds = tf.data.Dataset.zip((train_ds_one, train_ds_two))
+
+test_ds = (
+    test_ds.map(preprocess_image, num_parallel_calls=AUTO)
+    .batch(BATCH_SIZE)
+    .prefetch(AUTO)
+)
+
+"""
+##Define the CutMix Technique Function
+
+cutmix function takes two `image` and `label` pair and performs cutmix augmentation. It
+samples `λ(l)` from a [Beta
+Distribution](https://en.wikipedia.org/wiki/Beta_distribution) and we get the bounding
+box from get_box function. We crop the image2 and pad this image in the final padded
+image at the same location.
+"""
+
+
+@tf.function
+def get_box(l):
+    cut_rat = tf.math.sqrt(1.0 - l)
+
+    cut_w = IMG_SHAPE * cut_rat  # rw
+    cut_w = tf.cast(cut_w, tf.int32)
+
+    cut_h = IMG_SHAPE * cut_rat  # rh
+    cut_h = tf.cast(cut_h, tf.int32)
+
+    cx = tf.random.uniform((1,), minval=0, maxval=IMG_SHAPE, dtype=tf.int32)  # rx
+    cy = tf.random.uniform((1,), minval=0, maxval=IMG_SHAPE, dtype=tf.int32)  # ry
+
+    bbx1 = tf.clip_by_value(cx[0] - cut_w // 2, 0, IMG_SHAPE)
+    bby1 = tf.clip_by_value(cy[0] - cut_h // 2, 0, IMG_SHAPE)
+    bbx2 = tf.clip_by_value(cx[0] + cut_w // 2, 0, IMG_SHAPE)
+    bby2 = tf.clip_by_value(cy[0] + cut_h // 2, 0, IMG_SHAPE)
+
+    target_h = bby2 - bby1
+    if target_h == 0:
+        target_h += 1
+
+    target_w = bbx2 - bbx1
+    if target_w == 0:
+        target_w += 1
+
+    return bbx1, bby1, target_h, target_w
+
+
+@tf.function
+def cutmix(a, b):
+
+    (image1, label1), (image2, label2) = a, b
+
+    alpha = [1.0]
+    beta = [1.0]
+
+    ## Get sample from beta distribution
+    dist = tfd.Beta(alpha, beta)
```

Comment:
Please remove the usage of TF Probability and instead write a beta sampling function in TF like here: https://keras.io/examples/vision/mixup/

Reply:
Sure! Try to implement this here.